### DISCUSSION TITLE
How to minimize I/O while sending files

### DISCUSSION THREAD

**Discussion by watn3y** [go-telegram-bot-api/telegram-bot-api#526] [Q&A] [answered]
How would I go about minimizing I/O when uploading a file? I.e. send/upload as little data as possible.

would 

```
sendByCopy := tgbotapi.NewCopyMessage(chatID, fromChatID, messageID)
bot.CopyMessage(sendByCopy)
```
or
```
sendByFileID:= tgbotapi.NewDocument(chatID, tgbotapi.FileID(fileID))
bot.Send(sendByFileID)
```

be most efficient?

**Comment by Syfaro** [comment, marked as answer]
Either of those would be pretty similar in terms of reducing I/O for the bot (both are sending fairly small requests with some kind of ID), I don't think there's any way to use less resources to upload an existing file.

**Reply by watn3y** [reply]
Thanks :)
